### PR TITLE
show federation expiration metadata

### PR DESCRIPTION
**Pull request by futurepaul**
for #1128 

here's what I have so far. putting this here so @TonyGiorgio can see the logic I used to get the metadata

Screenshot 2024-05-07 at 10 22 03 AM

### CODE REVIEW

**Comment by TonyGiorgio** [comment]
1. It's not obvious what I'm supposed to do after the "Manage Federations" pop up window comes up. I think if the "transfer funds" button stayed on the federation page no matter what, it would be more obvious. A popup could come up, saying whether they want to swap to lightning or swap to a new federation and they should add one if they have it?

2. When the federation is the 2nd one, the pop up doesn't happen.

**Comment by futurepaul** [comment]
needs https://github.com/MutinyWallet/mutiny-node/pull/1169 and a new version of mutiny node

File: src/routes/Transfer.tsx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    const [state, _actions, sw] = useMegaStore();
+    const i18n = useI18n();
+    const navigate = useNavigate();
+    const [amountSats, setAmountSats] = createSignal(0n);
+    const [loading, setLoading] = createSignal(false);
+    const [params] = useSearchParams();
+
+    const canTransfer = createMemo(() => {
+        return true;
+    });
```
Comment (benthecarman) [inline src/routes/Transfer.tsx:41]: this doesn't seem useful

File: src/routes/Transfer.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        } catch (e) {
+            const error = eify(e);
+            setTransferResult({ failure_reason: error.message });
+            console.error(e);
+        } finally {
+            setLoading(false);
+        }
+    }
+
+    // const fromFederatationId = params.from;
```
Comment (benthecarman) [inline src/routes/Transfer.tsx:114]: ping

**Review by TonyGiorgio — APPROVED** [review]

File: src/components/AmountEditable.tsx [hunk 3]
```diff
@@ -29,6 +29,8 @@ function methodToIcon(method: MethodChoice["method"]) {
         return "lightning";
     } else if (method === "onchain") {
         return "chain";
+    } else if (method === "fedimint") {
+        return "community";
```
Comment (benthecarman) [inline src/components/AmountEditable.tsx:33]: you have a comment here saying to add fedimint, you can remove that now.

also would be better as a switch than if statements

**Review by benthecarman — APPROVED** [review]

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]
Needs retest

**Comment by futurepaul** [comment]
looks good now @TonyGiorgio

**Review by TonyGiorgio — APPROVED** [review]
Works great, just a nit that the federation name in the popup would be useful.

**Pull request merged**